packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,545 @@
+import EmberObject from 'ember-runtime/system/object';
+import { A as emberA } from 'ember-runtime/system/native_array';
+
+moduleFor('Helpers test: {{unbound}}', class extends RenderingTest {
+
+  ['@htmlbars it should not update the value if the parent view doesn\'t rerender']() {
+    this.render(`{{unbound foo}} {{unbound bar}}`, {
+      foo: 'BORK',
+      barBinding: 'foo'
+    });
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@htmlbars should be able to output a property without binding']() {
+    this.render(`<div id="first">{{unbound content.anUnboundString}}</div>`, {
+      content: EmberObject.create({
+        anUnboundString: 'No spans here, son.'
+      })
+    });
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => set(this.context, 'content.anUnboundString', 'HEY'));
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => set(this.context, 'content', EmberObject.create({
+      anUnboundString: 'No spans here, son.'
+    })));
+
+    this.assertText('No spans here, son.');
+  }
+
+  ['@htmlbars should be able to use unbound helper in #each helper']() {
+    this.render(`<ul>{{#each view.items as |item|}}<li>{{unbound item}}</li>{{/each}}</ul>`, {
+      items: emberA(['a', 'b', 'c', 1, 2, 3])
+    });
+
+    this.assertText('abc123');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc123');
+
+    this.runTask(() => set(this.context.items, 'firstObject', 'hello'));
+
+    this.assertText('abc123');
+
+    this.runTask(() => set(this.context, 'items', emberA(['a', 'b', 'c', 1, 2, 3])));
+
+    this.assertText('abc123');
+  }
+
+  ['@htmlbars should be able to use unbound helper in #each helper (with objects)']() {
+    this.render(`<ul>{{#each view.items as |item|}}<li>{{unbound item.wham}}</li>{{/each}}</ul>`, {
+      items: emberA([{ wham: 'bam' }, { wham: 1 }])
+    });
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.context.items.setEach('wham', 'HEY'));
+
+    this.assertText('bam1');
+
+    this.runTask(() => set(this.context, 'items',  emberA([{ wham: 'bam' }, { wham: 1 }])));
+
+    this.assertText('bam1');
+  }
+
+  ['@htmlbars it should assert unbound cannot be called with multiple arguments']({ throws }) {
+    let willThrow = () => {
+      this.render(`{{unbound foo bar}}`, {
+        foo: 'BORK',
+        bar: 'BLOOP'
+      });
+    };
+
+    throws(willThrow, /unbound helper cannot be called with multiple params or hash params/);
+  }
+
+  ['@htmlbars should render on attributes']() {
```

Comment:
Port of https://github.com/emberjs/ember.js/blob/bf067f7370c3c3381352d68ef5dfe55d1178c6f7/packages/ember-htmlbars/tests/helpers/unbound_test.js#L106-L122